From 32b863fade0e4d49ebe3838d1c54cbbf4aa17d7a Mon Sep 17 00:00:00 2001
From: Erik Axelsson <erik.axelsson@flir.com>
Date: Fri, 16 Oct 2020 13:04:13 +0200
Subject: [PATCH] SHLK-2505: Change PMIC hard reset time to 4s. This is done
 both in Linux and U-boot to make sure we catch all cases, even those where
 U-boot won't be updated

---
 drivers/mfd/pf1550.c       | 12 ++++++++++++
 include/linux/mfd/pf1550.h |  7 +++++++
 2 files changed, 19 insertions(+)

diff --git a/drivers/mfd/pf1550.c b/drivers/mfd/pf1550.c
index 44c92f466a99..ff7a51cf0dd7 100644
--- a/drivers/mfd/pf1550.c
+++ b/drivers/mfd/pf1550.c
@@ -173,6 +173,18 @@ static int pf1550_i2c_probe(struct i2c_client *i2c,
 		return ret;
 	}
 
+	ret = regmap_read(pf1550->regmap, PF1550_PMIC_REG_PWRCTRL0, &reg_data);
+	if (ret < 0) {
+		dev_err(pf1550->dev, "Could not read PWCTRL0 reg\n");
+		return ret;
+	}
+	reg_data = (reg_data & ~PF1550_PMIC_REG_PWRCTRL0_TGRESET_MASK) | PF1550_PMIC_REG_PWRCTRL0_TGRESET_4S;
+	ret = regmap_write(pf1550->regmap, PF1550_PMIC_REG_PWRCTRL0, reg_data);
+	if (ret < 0) {
+		dev_err(pf1550->dev, "Could not write PWCTRL0 reg\n");
+		return ret;
+	}
+
 	pf1550->type = PF1550;
 	dev_info(pf1550->dev, "pf1550 found.\n");
 
diff --git a/include/linux/mfd/pf1550.h b/include/linux/mfd/pf1550.h
index ac490196fd2c..34884f4ce5db 100644
--- a/include/linux/mfd/pf1550.h
+++ b/include/linux/mfd/pf1550.h
@@ -166,6 +166,13 @@ enum pf1550_pmic_reg {
 #define PF1550_CHARG_REG_VBUS_INLIM_CNFG_MASK 0xf8
 #define PF1550_CHARG_REG_VBUS_INLIM_CNFG_SHIFT 3
 
+
+#define PF1550_PMIC_REG_PWRCTRL0_TGRESET_MASK		(0x3 << 6)
+#define PF1550_PMIC_REG_PWRCTRL0_TGRESET_4S 		(0x0 << 6)
+#define PF1550_PMIC_REG_PWRCTRL0_TGRESET_8S 		(0x1 << 6)
+#define PF1550_PMIC_REG_PWRCTRL0_TGRESET_12S		(0x2 << 6)
+#define PF1550_PMIC_REG_PWRCTRL0_TGRESET_16S		(0x3 << 6)
+
 #define PF1550_CHARG_REG_CHG_OPER_CHG_OPER_MASK 0x3
 
 enum pf1550_chg_oper{
-- 
2.17.1

